keytool/src/main/scala/org/constellation/keytool/KeyStoreUtils.scala

```diff
@@ -25,6 +26,38 @@ object KeyStoreUtils {
       new FileOutputStream(file)
     })
 
+  def readFromFileStream[F[_]: Sync, T](dataPath: String, streamParser: FileInputStream => F[T]) =
+    reader(dataPath)
+      .use(streamParser)
+      .attemptT
+
+  def storeWithFileStream[F[_]: Sync](path: String, bufferedWriter: FileOutputStream => F[Unit]) =
+    writer(path)
+      .use(bufferedWriter)
+      .attemptT
+
+  def parseFileOfTypeOp[F[_]: Sync, T](parser: String => T)(stream: FileInputStream) =
+    Resource
+      .fromAutoCloseable(Sync[F].delay {
+        new BufferedReader(new InputStreamReader(stream))
+      })
+      .use(inStream => Option(inStream.readLine()).map(parser).pure[F])
+
+  def writeTypeToFileStream[F[_]: Sync, T](serializer: T => String)(obj: T)(stream: FileOutputStream) =
+    Resource
+      .fromAutoCloseable(Sync[F].delay {
+        new BufferedWriter(new OutputStreamWriter(stream))
+      })
+      .use(outStream => outStream.write(serializer(obj)).pure[F])
+
+  def storeKeyPemDecrypted[F[_]: Sync](key: Key)(stream: FileOutputStream): F[Unit] =
+    Resource
+      .fromAutoCloseable(Sync[F].delay { new PemWriter(new OutputStreamWriter(stream)) })
+      .use { pemWriter =>
+        val pemObj = new PemObject("EC KEY", key.getEncoded)
+        pemWriter.writeObject(pemObj).pure[F]
```

Comment:
is the write operation a pure one? I think it should be `Sync[F].delay { pemWriter.writeObject(pemObj) }` for all of these write statements